test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs

```diff
@@ -339,6 +340,71 @@ IConnection ConnectionFactory()
                 });
             }
 
+            [Fact]
+            public async Task ClientTimesoutWhenHandshakeResponseTakesTooLong()
+            {
+                var connection = new TestConnection(autoHandshake: false);
+                var hubConnection = CreateHubConnection(connection);
+                try
+                {
+                    hubConnection.HandshakeTimeout = TimeSpan.FromMilliseconds(1);
+
+                    await Assert.ThrowsAsync<OperationCanceledException>(() => hubConnection.StartAsync().OrTimeout());
+                }
+                finally
+                {
+                    await hubConnection.DisposeAsync().OrTimeout();
+                    await connection.DisposeAsync().OrTimeout();
+                }
+            }
+
+            [Fact]
+            public async Task StartAsyncWithTriggeredCancellationTokenIsCanceled()
+            {
+                var onStartCalled = false;
+                var connection = new TestConnection(onStart: () =>
+                {
+                    onStartCalled = true;
+                    return Task.CompletedTask;
+                });
+                var hubConnection = CreateHubConnection(connection);
+                try
+                {
+                    await Assert.ThrowsAsync<OperationCanceledException>(() => hubConnection.StartAsync(new CancellationToken(canceled: true)).OrTimeout());
+                    Assert.False(onStartCalled);
+                }
+                finally
+                {
+                    await hubConnection.DisposeAsync().OrTimeout();
+                    await connection.DisposeAsync().OrTimeout();
+                }
+            }
+
+            [Fact]
+            public async Task StartAsyncCanTriggerCancellationTokenToCancelHandshake()
+            {
+                var cts = new CancellationTokenSource();
+                var connection = new TestConnection(onStart: () =>
+                {
+                    cts.Cancel();
+                    return Task.CompletedTask;
+                }, autoHandshake: false);
+                var hubConnection = CreateHubConnection(connection);
+                // We want to make sure the cancellation is because of the token passed to StartAsync
+                hubConnection.HandshakeTimeout = TimeSpan.FromSeconds(20);
```

Comment:
Set it to `Timeout.InfiniteTimespan` then. Then the handshake timeout will never fire.